The job **`x86_64-gnu-llvm-19`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14494192187/job/40657639352) [(plain)](https://github.com/rust-lang/rust/commit/cf551dbca4ba89551b6e2c90fa03d888b4f3b304/checks/40657639352/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#19 exporting to docker image format
#19 sending tarball 25.9s done
#19 DONE 32.9s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-19]
[CI_JOB_NAME=x86_64-gnu-llvm-19]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Listening on address 127.0.0.1:4226
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-19', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'build.print-step-timings', '--enable-verbose-tests', '--set', 'build.metrics', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: build.build          := x86_64-unknown-linux-gnu
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-19/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
  Number of decisions:   4447
  longest path:          1159 (code:    152)
  longest backtrack:       66 (code:    428)
Shared 86733 out of 152951 states by creating 14756 new states, saving 71977
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/expmed.cc: In function ‘rtx_def* extract_bit_field_1(rtx, poly_uint64, poly_uint64, int, rtx, machine_mode, machine_mode, bool, bool, rtx_def**)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/expmed.cc:1864:45: warning: ‘*(unsigned int*)((char*)&imode + offsetof(scalar_int_mode, scalar_int_mode::m_mode))’ may be used uninitialized [-Wmaybe-uninitialized]
 1864 |       rtx sub = extract_bit_field_as_subreg (mode1, op0, imode,
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
 1865 |                                              bitsize, bitnum);
      |                                              ~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/expmed.cc:1824:19: note: ‘*(unsigned int*)((char*)&imode + offsetof(scalar_int_mode, scalar_int_mode::m_mode))’ was declared here
 1824 |   scalar_int_mode imode;
      |                   ^~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gimple-range-gori.cc: In member function ‘void range_def_chain::dump(FILE*, basic_block, const char*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gimple-range-gori.cc:319:19: warning: format not a string literal and no format arguments [-Wformat-security]
  319 |           fprintf (f, prefix);
      |           ~~~~~~~~^~~~~~~~~~~
---
                 from /checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/analyzer/region-model.h:33,
                 from /checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/analyzer/access-diagram.cc:39:
In constructor ‘ana::byte_range::byte_range(ana::byte_offset_t, ana::byte_size_t)’,
    inlined from ‘virtual text_art::table ana::string_literal_spatial_item::make_table(const ana::bit_to_table_map&, text_art::style_manager&) const’ at /checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/analyzer/access-diagram.cc:1812:18:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/analyzer/store.h:312:5: warning: ‘size_in_bytes.generic_wide_int<fixed_wide_int_storage<128> >::fixed_wide_int_storage<128>.fixed_wide_int_storage<128>::val[1]’ may be used uninitialized [-Wmaybe-uninitialized]
  312 |     m_size_in_bytes (size_in_bytes)
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/analyzer/access-diagram.cc: In member function ‘virtual text_art::table ana::string_literal_spatial_item::make_table(const ana::bit_to_table_map&, text_art::style_manager&) const’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/analyzer/access-diagram.cc:1808:28: note: ‘size_in_bytes.generic_wide_int<fixed_wide_int_storage<128> >::fixed_wide_int_storage<128>.fixed_wide_int_storage<128>::val[1]’ was declared here
 1808 |                byte_size_t size_in_bytes
      |                            ^~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/diagnostic.cc: In function ‘void fancy_abort(const char*, int, const char*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/diagnostic.cc:1677:15: warning: format not a string literal and no format arguments [-Wformat-security]
 1677 |       fnotice (stderr, diagnostic_kind_text[DK_ICE]);
---
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gcc.cc:7930:9: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 7930 |   write (fd, "\n\n", 2);
      |   ~~~~~~^~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gcc.cc: In member function ‘void driver::final_actions() const’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gcc.cc:9307:13: warning: ignoring return value of ‘int truncate(const char*, __off_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 9307 |     truncate(totruncate_file, 0);
      |     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In file included from /checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/jit/jit-playback.h:31,
                 from /checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/jit/jit-builtins.cc:24:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/jit/jit-recording.h: In member function ‘virtual bool gcc::jit::recording::type::is_same_type_as(gcc::jit::recording::type*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/jit/jit-recording.h:640:20: warning: suggest parentheses around ‘&&’ within ‘||’ [-Wparentheses]
---
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc: In function ‘void lto_resolution_read(splay_tree, FILE*, lto_file*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc:2091:10: warning: ignoring return value of ‘int fscanf(FILE*, const char*, ...)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2091 |   fscanf (resolution, " ");   /* Read white space.  */
      |   ~~~~~~~^~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc:2093:9: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2093 |   fread (obj_name, sizeof (char), name_len, resolution);
      |   ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc:2113:10: warning: ignoring return value of ‘int fscanf(FILE*, const char*, ...)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2113 |   fscanf (resolution, "%u", &num_symbols);
      |   ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/jit/jit-recording.cc:32:
---
Applying io_quotes_use            to linux/blkzoned.h
Applying io_quotes_use            to linux/ipmi.h
Applying io_quotes_use            to linux/psp-dbc.h
Applying io_quotes_use            to linux/bt-bmc.h
Applying io_quotes_use            to linux/tps6594_pfsm.h
Applying io_quotes_use            to linux/cxl_mem.h
Applying io_quotes_use            to linux/wmi.h
Applying io_quotes_use            to linux/auto_fs.h
Applying io_quotes_use            to linux/mmtimer.h
Applying io_quotes_use            to linux/f2fs.h
Applying io_quotes_use            to linux/vhost.h
---
Applying io_quotes_use            to sound/asound.h
Applying io_quotes_use            to sound/compress_offload.h
Applying hpux8_bogus_inlines      to math.h
Applying pthread_incomplete_struct_argument to pthread.h
Fixed:  pthread.h
Applying io_quotes_use            to misc/mrvl_cn10k_dpi.h
Applying io_quotes_use            to misc/ocxl.h
Applying io_quotes_use            to misc/cxl.h
Applying io_quotes_use            to misc/xilinx_sdfec.h
Applying io_quotes_def            to unicode/platform.h
Applying sun_malloc               to malloc.h
Applying io_quotes_use            to scsi/cxlflash_ioctl.h
---
Applying machine_name             to x86_64-linux-gnu/bits/unistd_ext.h
Applying io_quotes_use            to x86_64-linux-gnu/asm/mtrr.h
Applying io_quotes_use            to x86_64-linux-gnu/asm/amd_hsmp.h
Applying machine_name             to openssl/e_os2.h
Applying io_quotes_use            to drm/xe_drm.h
Applying io_quotes_use            to drm/radeon_drm.h
Applying io_quotes_use            to drm/panfrost_drm.h
Applying io_quotes_use            to drm/etnaviv_drm.h
Applying io_quotes_use            to drm/lima_drm.h
Applying io_quotes_use            to drm/qaic_accel.h
Applying io_quotes_use            to drm/vc4_drm.h
Applying io_quotes_use            to drm/i915_drm.h
Applying io_quotes_use            to drm/omap_drm.h
Applying io_quotes_use            to drm/pvr_drm.h
Applying io_quotes_use            to drm/amdgpu_drm.h
Applying io_quotes_use            to drm/vgem_drm.h
Applying io_quotes_use            to drm/msm_drm.h
Applying io_quotes_use            to drm/v3d_drm.h
Applying io_quotes_use            to drm/exynos_drm.h
Applying io_quotes_use            to drm/nouveau_drm.h
Applying io_quotes_use            to drm/drm.h
Applying io_quotes_use            to drm/habanalabs_accel.h
Applying io_quotes_use            to drm/tegra_drm.h
Applying io_quotes_use            to rdma/rdma_user_ioctl.h
cc1: note: self-tests are not enabled in this build
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/c++tools/server.cc: In function ‘void server(bool, int, module_resolver*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/c++tools/server.cc:620:10: warning: ignoring return value of ‘int pipe(int*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
---
diff of stderr:

2   --> $DIR/issue-11004.rs:7:21
3    |
4 LL |     let x : i32 = n.x;
-    |                     ^ unknown field
+    |                   --^
+    |                   |
+    |                   unknown field access via raw pointer
6    |
- help: `n` is a raw pointer; try dereferencing it
+ help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
8    |
- LL |     let x : i32 = (*n).x;
-    |                   ++ +
+ LL -     let x : i32 = n.x;
+ LL +     let x : i32 = (*n).x;
+    |
11 
12 error[E0609]: no field `y` on type `*mut A`
13   --> $DIR/issue-11004.rs:8:21

14    |
15 LL |     let y : f64 = n.y;
-    |                     ^ unknown field
+    |                   --^
+    |                   |
+    |                   unknown field access via raw pointer
17    |
- help: `n` is a raw pointer; try dereferencing it
+ help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
19    |
- LL |     let y : f64 = (*n).y;
-    |                   ++ +
+ LL -     let y : f64 = n.y;
+ LL +     let y : f64 = (*n).y;
+    |
22 
23 error: aborting due to 2 previous errors
24 

Note: some mismatched output was normalized before being compared
- LL -     let x : i32 = n.x; //~ ERROR no field `x` on type `*mut A`
- LL +     let x : i32 = (*n).x; //~ ERROR no field `x` on type `*mut A`
- LL -     let y : f64 = n.y; //~ ERROR no field `y` on type `*mut A`
- LL +     let y : f64 = (*n).y; //~ ERROR no field `y` on type `*mut A`
+    |                   --^
+    |                   |
+    |                   unknown field access via raw pointer
+ help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
+ LL -     let x : i32 = n.x;
+ LL +     let x : i32 = (*n).x;
+    |
+    |                   --^
+    |                   |
+    |                   unknown field access via raw pointer
+ help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
+ LL -     let y : f64 = n.y;
+ LL +     let y : f64 = (*n).y;
+    |


The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args issues/issue-11004.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/issues/issue-11004.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-11004" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0609]: no field `x` on type `*mut A`
##[error]  --> /checkout/tests/ui/issues/issue-11004.rs:7:21
   |
LL |     let x : i32 = n.x; //~ ERROR no field `x` on type `*mut A`
   |                   --^
   |                   |
   |                   unknown field access via raw pointer
   |
help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
   |
LL -     let x : i32 = n.x; //~ ERROR no field `x` on type `*mut A`
LL +     let x : i32 = (*n).x; //~ ERROR no field `x` on type `*mut A`
   |

error[E0609]: no field `y` on type `*mut A`
##[error]  --> /checkout/tests/ui/issues/issue-11004.rs:8:21
   |
LL |     let y : f64 = n.y; //~ ERROR no field `y` on type `*mut A`
   |                   --^
   |                   |
   |                   unknown field access via raw pointer
   |
help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
   |
LL -     let y : f64 = n.y; //~ ERROR no field `y` on type `*mut A`
LL +     let y : f64 = (*n).y; //~ ERROR no field `y` on type `*mut A`
   |

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0609`.
------------------------------------------


---- [ui] tests/ui/parser/expr-rarrow-call.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/expr-rarrow-call/expr-rarrow-call.stderr"
diff of stderr:

4 LL |     named->foo;
5    |          ^^
6    |
-    = help: the `.` operator will dereference the value if needed
+    = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
8 help: try using `.` instead
9    |
10 LL -     named->foo;

17 LL |     unnamed->0;
18    |            ^^
19    |
-    = help: the `.` operator will dereference the value if needed
+    = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
21 help: try using `.` instead
22    |
23 LL -     unnamed->0;

30 LL |     t->0;
31    |      ^^
32    |
-    = help: the `.` operator will dereference the value if needed
+    = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
34 help: try using `.` instead
35    |
36 LL -     t->0;

43 LL |     t->1;
44    |      ^^
45    |
-    = help: the `.` operator will dereference the value if needed
+    = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
47 help: try using `.` instead
48    |
49 LL -     t->1;

56 LL |     foo->clone();
57    |        ^^
58    |
-    = help: the `.` operator will dereference the value if needed
+    = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
60 help: try using `.` instead
61    |
62 LL -     foo->clone();


The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args parser/expr-rarrow-call.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parser/expr-rarrow-call.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/expr-rarrow-call" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: `->` used for field access or method call
##[error]  --> /checkout/tests/ui/parser/expr-rarrow-call.rs:14:10
   |
LL |     named->foo; //~ ERROR `->` used for field access or method call
   |          ^^
   |
   = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
help: try using `.` instead
   |
LL -     named->foo; //~ ERROR `->` used for field access or method call
LL +     named.foo; //~ ERROR `->` used for field access or method call
   |

error: `->` used for field access or method call
##[error]  --> /checkout/tests/ui/parser/expr-rarrow-call.rs:18:12
   |
LL |     unnamed->0; //~ ERROR `->` used for field access or method call
   |            ^^
   |
   = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
help: try using `.` instead
   |
LL -     unnamed->0; //~ ERROR `->` used for field access or method call
LL +     unnamed.0; //~ ERROR `->` used for field access or method call
   |

error: `->` used for field access or method call
##[error]  --> /checkout/tests/ui/parser/expr-rarrow-call.rs:22:6
   |
LL |     t->0; //~ ERROR `->` used for field access or method call
   |      ^^
   |
   = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
help: try using `.` instead
   |
LL -     t->0; //~ ERROR `->` used for field access or method call
LL +     t.0; //~ ERROR `->` used for field access or method call
   |

error: `->` used for field access or method call
##[error]  --> /checkout/tests/ui/parser/expr-rarrow-call.rs:23:6
   |
LL |     t->1; //~ ERROR `->` used for field access or method call
   |      ^^
   |
   = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
help: try using `.` instead
   |
LL -     t->1; //~ ERROR `->` used for field access or method call
LL +     t.1; //~ ERROR `->` used for field access or method call
   |

error: `->` used for field access or method call
##[error]  --> /checkout/tests/ui/parser/expr-rarrow-call.rs:30:8
   |
LL |     foo->clone(); //~ ERROR `->` used for field access or method call
   |        ^^
   |
   = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
help: try using `.` instead
   |
LL -     foo->clone(); //~ ERROR `->` used for field access or method call
LL +     foo.clone(); //~ ERROR `->` used for field access or method call
   |

error: aborting due to 5 previous errors
------------------------------------------

---

39 LL |     attr::fn bar() -> String {
40    |                    ^^
41    |
-    = help: the `.` operator will dereference the value if needed
+    = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
43 help: try using `.` instead
44    |
45 LL -     attr::fn bar() -> String {


The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args parser/issues/issue-118530-ice.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parser/issues/issue-118530-ice.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/issues/issue-118530-ice" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: expected `;`, found `#`
##[error]  --> /checkout/tests/ui/parser/issues/issue-118530-ice.rs:3:21
   |
LL |     #[cfg]
   |     ------ only `;` terminated statements or tail expressions are allowed after this attribute
LL |     [1, 2, 3].iter() //~ ERROR expected `;`, found `#`
   |                     ^ expected `;` here
LL |     #[feature]
   |     - unexpected token
   |
help: add `;` here
   |
LL |     [1, 2, 3].iter(); //~ ERROR expected `;`, found `#`
   |                     +
help: alternatively, consider surrounding the expression with a block
   |
LL |     { [1, 2, 3].iter() } //~ ERROR expected `;`, found `#`
   |     +                  +

error: expected identifier, found keyword `fn`
##[error]  --> /checkout/tests/ui/parser/issues/issue-118530-ice.rs:5:11
   |
LL |     attr::fn bar() -> String { //~ ERROR expected identifier, found keyword `fn`
   |           ^^ expected identifier, found keyword

error: expected `;`, found `bar`
##[error]  --> /checkout/tests/ui/parser/issues/issue-118530-ice.rs:5:13
   |
LL |     #[feature]
   |     ---------- only `;` terminated statements or tail expressions are allowed after this attribute
LL |     attr::fn bar() -> String { //~ ERROR expected identifier, found keyword `fn`
   |             ^--- unexpected token
   |             |
   |             help: add `;` here

error: `->` used for field access or method call
##[error]  --> /checkout/tests/ui/parser/issues/issue-118530-ice.rs:5:20
   |
LL |     attr::fn bar() -> String { //~ ERROR expected identifier, found keyword `fn`
   |                    ^^
   |
   = help: the `.` operator will dereference the value if needed (if value is raw pointer you have to derefernce it by yourself)
help: try using `.` instead
   |
LL -     attr::fn bar() -> String { //~ ERROR expected identifier, found keyword `fn`
LL +     attr::fn bar() . String { //~ ERROR expected identifier, found keyword `fn`
   |

error: expected one of `(`, `.`, `::`, `;`, `?`, `}`, or an operator, found `{`
##[error]  --> /checkout/tests/ui/parser/issues/issue-118530-ice.rs:5:30
   |
LL |     attr::fn bar() -> String { //~ ERROR expected identifier, found keyword `fn`
   |                              ^ expected one of 7 possible tokens

error: aborting due to 5 previous errors
------------------------------------------


---- [ui] tests/ui/suggestions/parenthesized-deref-suggestion.rs stdout ----
Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/suggestions/parenthesized-deref-suggestion/parenthesized-deref-suggestion.stderr"
diff of stderr:

2   --> $DIR/parenthesized-deref-suggestion.rs:7:30
3    |
4 LL |     (sess as *const Session).opts;
-    |                              ^^^^ unknown field
+    |     -------------------------^^^^
+    |     |
+    |     unknown field access via raw pointer
6    |
- help: the value is a raw pointer; try dereferencing it
+ help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
8    |
- LL |     (*(sess as *const Session)).opts;
-    |     ++                        +
+ LL -     (sess as *const Session).opts;
+ LL +     (*(sess as *const Session)).opts;
+    |
11 
12 error[E0609]: no field `0` on type `[u32; 1]`
13   --> $DIR/parenthesized-deref-suggestion.rs:10:21

Note: some mismatched output was normalized before being compared
- LL -     (sess as *const Session).opts; //~ ERROR no field `opts` on type `*const Session`
- LL +     (*(sess as *const Session)).opts; //~ ERROR no field `opts` on type `*const Session`
+    |     -------------------------^^^^
+    |     |
+    |     unknown field access via raw pointer
+ help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
+ LL -     (sess as *const Session).opts;
+ LL +     (*(sess as *const Session)).opts;
+    |


The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args suggestions/parenthesized-deref-suggestion.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/suggestions/parenthesized-deref-suggestion.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/suggestions/parenthesized-deref-suggestion" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0609]: no field `opts` on type `*const Session`
##[error]  --> /checkout/tests/ui/suggestions/parenthesized-deref-suggestion.rs:7:30
   |
LL |     (sess as *const Session).opts; //~ ERROR no field `opts` on type `*const Session`
   |     -------------------------^^^^
   |     |
   |     unknown field access via raw pointer
   |
help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
   |
LL -     (sess as *const Session).opts; //~ ERROR no field `opts` on type `*const Session`
LL +     (*(sess as *const Session)).opts; //~ ERROR no field `opts` on type `*const Session`
   |

error[E0609]: no field `0` on type `[u32; 1]`
##[error]  --> /checkout/tests/ui/suggestions/parenthesized-deref-suggestion.rs:10:21
   |
LL |     (x as [u32; 1]).0; //~ ERROR no field `0` on type `[u32; 1]`
   |                     ^ unknown field
   |
help: instead of using tuple indexing, use array indexing
   |
LL -     (x as [u32; 1]).0; //~ ERROR no field `0` on type `[u32; 1]`
LL +     (x as [u32; 1])[0]; //~ ERROR no field `0` on type `[u32; 1]`
   |

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0609`.
---
+    |            --^
+    |            |
+    |            unknown field access via raw pointer
6    |
- help: `p` is a raw pointer; try dereferencing it
+ help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
8    |
- LL |     return (*p).f;
-    |            ++ +
+ LL -     return p.f;
+ LL +     return (*p).f;
+    |
11 
12 error: aborting due to 1 previous error
13 

Note: some mismatched output was normalized before being compared
- LL -     return p.f; //~ ERROR no field `f` on type `*const Rec`
- LL +     return (*p).f; //~ ERROR no field `f` on type `*const Rec`
+    |            --^
+    |            |
+    |            unknown field access via raw pointer
+ help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
+ LL -     return p.f;
+ LL +     return (*p).f;
+    |


The actual stderr differed from the expected stderr.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args unsafe/unsafe-fn-autoderef.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/unsafe/unsafe-fn-autoderef.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/unsafe/unsafe-fn-autoderef" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0609]: no field `f` on type `*const Rec`
##[error]  --> /checkout/tests/ui/unsafe/unsafe-fn-autoderef.rs:19:14
   |
LL |     return p.f; //~ ERROR no field `f` on type `*const Rec`
   |            --^
   |            |
   |            unknown field access via raw pointer
   |
help: `->` is not valid in Rust; use `.` on a dereferenced raw pointer instead
   |
LL -     return p.f; //~ ERROR no field `f` on type `*const Rec`
LL +     return (*p).f; //~ ERROR no field `f` on type `*const Rec`
   |

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0609`.
```

</details>

--- a/compiler/rustc_hir_typeck/src/expr.rs
+++ b/compiler/rustc_hir_typeck/src/expr.rs
@@ -3289,8 +3289,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         err.multipart_suggestion(
             format!("{val} is a raw pointer; try dereferencing it"),
             vec![
-                (base.span.shrink_to_lo(), "(*".to_string()),
-                (base.span.shrink_to_hi(), ")".to_string()),
+                (base.span.shrink_to_lo(), "(*".into()),
+                (base.span.between(field.span), format!(").")),
             ],
             Applicability::MaybeIncorrect,
         );

--- a/compiler/rustc_parse/messages.ftl
+++ b/compiler/rustc_parse/messages.ftl
@@ -246,9 +246,9 @@ parse_expected_struct_field = expected one of `,`, `:`, or `{"}"}`, found `{$tok
 
 parse_expected_trait_in_trait_impl_found_type = expected a trait, found type
 
-parse_expr_rarrow_call = `->` used for field access or method call
+parse_expr_rarrow_call = `->` is not valid syntax for field accesses and method calls
     .suggestion = try using `.` instead
-    .help = the `.` operator will dereference the value if needed
+    .help = the `.` operator will automatically dereference the value, except if the value is a raw pointer
 
 parse_extern_crate_name_with_dashes = crate name using dashes are not valid in `extern crate` statements
     .label = dash-separated idents are not valid

--- a/tests/ui/parser/expr-rarrow-call-on-a-raw-pointer.rs
+++ b/tests/ui/parser/expr-rarrow-call-on-a-raw-pointer.rs
@@ -1,0 +1,22 @@
+#![allow(
+    dead_code,
+    unused_must_use
+)]
+
+struct Named {
+    foo: usize,
+}
+
+struct Unnamed(usize);
+
+unsafe fn named_struct_field_access(named: *mut Named) {
+    named->foo += 1; //~ ERROR `->` is not valid syntax for field accesses and method calls
+    //~^ ERROR no field `foo` on type `*mut Named`
+}
+
+unsafe fn unnamed_struct_field_access(unnamed: *mut Unnamed) {
+    unnamed->0 += 1; //~ ERROR `->` is not valid syntax for field accesses and method calls
+    //~^ ERROR no field `0` on type `*mut Unnamed`
+}
+
+fn main() {}

--- a/tests/ui/parser/expr-rarrow-call-on-a-raw-pointer.stderr
+++ b/tests/ui/parser/expr-rarrow-call-on-a-raw-pointer.stderr
@@ -1,0 +1,53 @@
+error: `->` is not valid syntax for field accesses and method calls
+  --> $DIR/expr-rarrow-call-on-a-raw-pointer.rs:13:10
+   |
+LL |     named->foo += 1;
+   |          ^^
+   |
+   = help: the `.` operator will automatically dereference the value, except if the value is a raw pointer
+help: try using `.` instead
+   |
+LL -     named->foo += 1;
+LL +     named.foo += 1;
+   |
+
+error: `->` is not valid syntax for field accesses and method calls
+  --> $DIR/expr-rarrow-call-on-a-raw-pointer.rs:18:12
+   |
+LL |     unnamed->0 += 1;
+   |            ^^
+   |
+   = help: the `.` operator will automatically dereference the value, except if the value is a raw pointer
+help: try using `.` instead
+   |
+LL -     unnamed->0 += 1;
+LL +     unnamed.0 += 1;
+   |
+
+error[E0609]: no field `foo` on type `*mut Named`
+  --> $DIR/expr-rarrow-call-on-a-raw-pointer.rs:13:12
+   |
+LL |     named->foo += 1;
+   |            ^^^ unknown field
+   |
+help: `named` is a raw pointer; try dereferencing it
+   |
+LL -     named->foo += 1;
+LL +     (*named).foo += 1;
+   |
+
+error[E0609]: no field `0` on type `*mut Unnamed`
+  --> $DIR/expr-rarrow-call-on-a-raw-pointer.rs:18:14
+   |
+LL |     unnamed->0 += 1;
+   |              ^ unknown field
+   |
+help: `unnamed` is a raw pointer; try dereferencing it
+   |
+LL -     unnamed->0 += 1;
+LL +     (*unnamed).0 += 1;
+   |
+
+error: aborting due to 4 previous errors
+
+For more information about this error, try `rustc --explain E0609`.

--- a/tests/ui/parser/expr-rarrow-call.fixed
+++ b/tests/ui/parser/expr-rarrow-call.fixed
@@ -11,23 +11,23 @@ struct Named {
 struct Unnamed(usize);
 
 fn named_struct_field_access(named: &Named) {
-    named.foo; //~ ERROR `->` used for field access or method call
+    named.foo; //~ ERROR `->` is not valid syntax for field accesses and method calls
 }
 
 fn unnamed_struct_field_access(unnamed: &Unnamed) {
-    unnamed.0; //~ ERROR `->` used for field access or method call
+    unnamed.0; //~ ERROR `->` is not valid syntax for field accesses and method calls
 }
 
 fn tuple_field_access(t: &(u8, u8)) {
-    t.0; //~ ERROR `->` used for field access or method call
-    t.1; //~ ERROR `->` used for field access or method call
+    t.0; //~ ERROR `->` is not valid syntax for field accesses and method calls
+    t.1; //~ ERROR `->` is not valid syntax for field accesses and method calls
 }
 
 #[derive(Clone)]
 struct Foo;
 
 fn method_call(foo: &Foo) {
-    foo.clone(); //~ ERROR `->` used for field access or method call
+    foo.clone(); //~ ERROR `->` is not valid syntax for field accesses and method calls
 }
 
 fn main() {}

--- a/tests/ui/parser/expr-rarrow-call.rs
+++ b/tests/ui/parser/expr-rarrow-call.rs
@@ -11,23 +11,23 @@ struct Named {
 struct Unnamed(usize);
 
 fn named_struct_field_access(named: &Named) {
-    named->foo; //~ ERROR `->` used for field access or method call
+    named->foo; //~ ERROR `->` is not valid syntax for field accesses and method calls
 }
 
 fn unnamed_struct_field_access(unnamed: &Unnamed) {
-    unnamed->0; //~ ERROR `->` used for field access or method call
+    unnamed->0; //~ ERROR `->` is not valid syntax for field accesses and method calls
 }
 
 fn tuple_field_access(t: &(u8, u8)) {
-    t->0; //~ ERROR `->` used for field access or method call
-    t->1; //~ ERROR `->` used for field access or method call
+    t->0; //~ ERROR `->` is not valid syntax for field accesses and method calls
+    t->1; //~ ERROR `->` is not valid syntax for field accesses and method calls
 }
 
 #[derive(Clone)]
 struct Foo;
 
 fn method_call(foo: &Foo) {
-    foo->clone(); //~ ERROR `->` used for field access or method call
+    foo->clone(); //~ ERROR `->` is not valid syntax for field accesses and method calls
 }
 
 fn main() {}

--- a/tests/ui/parser/expr-rarrow-call.stderr
+++ b/tests/ui/parser/expr-rarrow-call.stderr
@@ -1,62 +1,62 @@
-error: `->` used for field access or method call
+error: `->` is not valid syntax for field accesses and method calls
   --> $DIR/expr-rarrow-call.rs:14:10
    |
 LL |     named->foo;
    |          ^^
    |
-   = help: the `.` operator will dereference the value if needed
+   = help: the `.` operator will automatically dereference the value, except if the value is a raw pointer
 help: try using `.` instead
    |
 LL -     named->foo;
 LL +     named.foo;
    |
 
-error: `->` used for field access or method call
+error: `->` is not valid syntax for field accesses and method calls
   --> $DIR/expr-rarrow-call.rs:18:12
    |
 LL |     unnamed->0;
    |            ^^
    |
-   = help: the `.` operator will dereference the value if needed
+   = help: the `.` operator will automatically dereference the value, except if the value is a raw pointer
 help: try using `.` instead
    |
 LL -     unnamed->0;
 LL +     unnamed.0;
    |
 
-error: `->` used for field access or method call
+error: `->` is not valid syntax for field accesses and method calls
   --> $DIR/expr-rarrow-call.rs:22:6
    |
 LL |     t->0;
    |      ^^
    |
-   = help: the `.` operator will dereference the value if needed
+   = help: the `.` operator will automatically dereference the value, except if the value is a raw pointer
 help: try using `.` instead
    |
 LL -     t->0;
 LL +     t.0;
    |
 
-error: `->` used for field access or method call
+error: `->` is not valid syntax for field accesses and method calls
   --> $DIR/expr-rarrow-call.rs:23:6
    |
 LL |     t->1;
    |      ^^
    |
-   = help: the `.` operator will dereference the value if needed
+   = help: the `.` operator will automatically dereference the value, except if the value is a raw pointer
 help: try using `.` instead
    |
 LL -     t->1;
 LL +     t.1;
    |
 
-error: `->` used for field access or method call
+error: `->` is not valid syntax for field accesses and method calls
   --> $DIR/expr-rarrow-call.rs:30:8
    |
 LL |     foo->clone();
    |        ^^
    |
-   = help: the `.` operator will dereference the value if needed
+   = help: the `.` operator will automatically dereference the value, except if the value is a raw pointer
 help: try using `.` instead
    |
 LL -     foo->clone();

--- a/tests/ui/parser/issues/issue-118530-ice.rs
+++ b/tests/ui/parser/issues/issue-118530-ice.rs
@@ -5,7 +5,7 @@ fn bar() -> String {
     attr::fn bar() -> String { //~ ERROR expected identifier, found keyword `fn`
     //~^ ERROR expected one of `(`, `.`, `::`, `;`, `?`, `}`, or an operator, found `{`
     //~| ERROR expected `;`, found `bar`
-    //~| ERROR `->` used for field access or method call
+    //~| ERROR `->` is not valid syntax for field accesses and method calls
     #[attr]
     [1, 2, 3].iter().map().collect::<String>()
     #[attr]

--- a/tests/ui/parser/issues/issue-118530-ice.stderr
+++ b/tests/ui/parser/issues/issue-118530-ice.stderr
@@ -33,13 +33,13 @@ LL |     attr::fn bar() -> String {
    |             |
    |             help: add `;` here
 
-error: `->` used for field access or method call
+error: `->` is not valid syntax for field accesses and method calls
   --> $DIR/issue-118530-ice.rs:5:20
    |
 LL |     attr::fn bar() -> String {
    |                    ^^
    |
-   = help: the `.` operator will dereference the value if needed
+   = help: the `.` operator will automatically dereference the value, except if the value is a raw pointer
 help: try using `.` instead
    |
 LL -     attr::fn bar() -> String {

